process/block/shardblock.go

```diff
@@ -1774,9 +1809,14 @@ func (sp *shardProcessor) computeExistingAndRequestMissingMetaHeaders(header dat
 		if hdr.Nonce > sp.hdrsForCurrBlock.highestHdrNonce[core.MetachainShardId] {
 			sp.hdrsForCurrBlock.highestHdrNonce[core.MetachainShardId] = hdr.Nonce
 		}
+		shouldConsiderProofsForNotarization := sp.enableEpochsHandler.IsFlagEnabledInEpoch(common.EquivalentMessagesFlag, hdr.Epoch)
```

Comment:
I don't think this is correct, as some of the referenced metablocks may be from the previous epoch.

Reply:
it should be covered with the current implementation, the number of notarizedMetaHdrsBasedOnProofs will not be equal with the len of metablockHashes so if there are no missing headers, it will request the finality attesting headers